### PR TITLE
Added basic PIC functionality

### PR DESCRIPTION
**Pull request by sidharth-sundar**
Added new PIC protocol in Simbricks with 128b global addresses, basic address translation, and 2 new simulators:

cpu_pic.c: Simulates the PIC connecting a CPU and a memory PIC
mem_pic.c: Simulates the PIC connecting the CPU PIC to the memory node

### CODE REVIEW

File: lib/simbricks/pic/if.h
Comment (pooriaPoorsarvi) [inline lib/simbricks/pic/if.h:1]: @sidharth-sundar plz remove all the copy right texts so we don't push our own code under it.

File: lib/simbricks/pic/if.c [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include "lib/simbricks/pic/if.h"
+
+void SimbricksPicIfDefaultParams(struct SimbricksBaseIfParams *params) {
+  SimbricksBaseIfDefaultParams(params);
+  params->upper_layer_proto = SIMBRICKS_PROTO_ID_PIC;
+  // fit DMA writes with size 8192
```
Comment (pooriaPoorsarvi) [inline lib/simbricks/pic/if.c:6]: can you explain this hard cded 8192 ?  @sidharth-sundar

Comment (sidharth-sundar) [reply lib/simbricks/pic/if.c:6]: I'm not sure on that, it's a remnant from the memory interface (lib/simbricks/mem/if.c). It's setting the amount of space required in shared memory, which is used for the input and output message queues. I think it's allocating enough space to fit an 8192B DMA write alongside the remainder of the packet body, but currently we don't issue such large writes.

File: lib/simbricks/pic/if.h
Comment (sidharth-sundar) [reply lib/simbricks/pic/if.h:1]: @pooriaPoorsarvi It should be fixed now.

**Comment by sidharth-sundar** [comment]
TODO: Push fixes to basicmem.c simulator upstream

**Pull request merged**